lib/npg_pipeline/validation.pm

```diff
@@ -670,13 +670,43 @@ sub _irods_seq_deletable {
   my $deletable = $v->archived_for_deletion();
   push @{$self->eligible_product_entities}, @{$v->eligible_product_entities};
 
+  # Stepping back from a convention to always run every check.
+  if ($deletable) {
+    $deletable = $self->_irods_seq_pp_deletable();
+  }
+
   my $m = sprintf 'Files in iRODS: run %i %sdeletable',
           $self->id_run , $deletable ? q[] : q[NOT ];
   $self->info($m);
 
   return $deletable;
 }
 
+sub _irods_seq_pp_deletable {
+  my $self = shift;
+  ######
+  # A simplified procedure, which will stop and return 0 (not deletable)
+  # as soon as something goes wrong or the first incorrectly archived
+  # file is found.
+  # No checks are done for products for which no pp was run, these
+  # products are considered deletable in the context of this method.
+  #
+  foreach my $p (@{$self->product_entities}) {
+    # get study conf for a product
+    # if 'irods_pp' > 'filters' > 'include' sectin exists
```

Comment:
I like the idea of using product release for picking the files. But you were right about it being very specific. I see the pp version is there. I remember it was possible to use a specific product release to manually execute the pipeline. Guess we can use something like that to delete older versions and use the current with a shorter staging_deletion_delay for more recent runs. Guess we will have a similar issue every time we switch versions but we can always address it manually until we have a general solution.